akka-remote/src/main/scala/akka/remote/artery/TcpFraming.scala

```diff
@@ -0,0 +1,141 @@
+/**
+ * Copyright (C) 2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.remote.artery
+
+import java.nio.ByteBuffer
+import java.nio.ByteOrder
+
+import scala.annotation.tailrec
+
+import akka.annotation.InternalApi
+import akka.stream.Attributes
+import akka.util.ByteString
+import akka.stream.FlowShape
+import akka.stream.Inlet
+import akka.stream.Outlet
+import akka.stream.stage.GraphStage
+import akka.stream.stage.GraphStageLogic
+import akka.stream.stage.InHandler
+import akka.stream.stage.OutHandler
+import akka.stream.stage.StageLogging
+import akka.remote.artery.FlightRecorderEvents.TcpInbound_Received
+import akka.stream.scaladsl.Framing.FramingException
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] object TcpFraming {
+  val Undefined = Int.MinValue
+
+  def frameHeader(size: Int) =
+    ByteString(
+      (size & 0xff).toByte,
+      ((size & 0xff00) >> 8).toByte,
+      ((size & 0xff0000) >> 16).toByte,
+      ((size & 0xff000000) >> 24).toByte
+    )
```

Comment:
Maybe move the frameLength decoding logic here as well for easy comparison?

Reply:
👍